include/posix/pthread.h

```diff
@@ -49,8 +49,8 @@ struct posix_thread {
 };
 
 /* Pthread detach/joinable */
-#define PTHREAD_CREATE_JOINABLE     0
-#define PTHREAD_CREATE_DETACHED     1
+#define PTHREAD_CREATE_JOINABLE     PTHREAD_JOINABLE
```

Comment:
This looks like a change unrelated to the purpose of the commit (as stated in the commit message)? Is it really needed? If not, I'd recommend to remove it. (And if it's really needed, the proper way would be to do it in a separate commit, with its message explaining why the change is done.)

Reply:
they will be used to init the state of struct pthread_state, so the value of PTHREAD_CREATE_JOINABLE and PTHREAD_CREATE_DETACHED should follow pthread_state enum.

Reply:
> they will be used to init the state of struct pthread_state

Sounds good, thanks.